### PR TITLE
Feat/q4 regular release

### PR DESCRIPTION
**Pull request by repjarms**
### Summary

This is the PR for the upcoming regular release. It adds the following:

- Support for DiscoveryV2 (Cloud Pak for Data only)
- Added Webhook support for AssistantV1
- Added support for Disambiguation opt out in AssistantV1
- General model updates for other services
- Model fixes for VisualRecognitionV4 (previously handled with hand edits)

Additionally, we updated the tests:

- Remove BasicAuthenticator support for tests (we don't support CF anymore so we use IAM for all tests now)
- Update tests for new test instances

NOTE: This PR is not to be merged until 12/2. Please feel free to review but do not merge in the meantime.

### CODE REVIEW

File: Source/DiscoveryV1/Models/QueryResult.swift [hunk 1]
```diff
@@ -42,11 +42,6 @@ public struct QueryResult: Codable, Equatable {
      */
     public var resultMetadata: QueryResultMetadata?
 
-    /**
-     Automatically extracted result title.
-     */
-    public var title: String?
-
```
Comment (mkistler) [inline Source/DiscoveryV1/Models/QueryResult.swift]: Why did `title` disappear from this model?  Technically this is a breaking change.

File: Source/DiscoveryV2/Models/QueryAggregation.swift [hunk 2]
```diff
@@ -0,0 +1,110 @@
+/**
+ <<<<<<< HEAD
+ * (C) Copyright IBM Corp. 2019.
+ =======
+ * (C) Copyright IBM Corp. 2018.
+ >>>>>>> 029bbf41... feat(hand edit): add discriminator support for QueryAggregation
```
Comment (mkistler) [inline Source/DiscoveryV2/Models/QueryAggregation.swift]: Need to fix your merge conflict here.

Comment (mediumTaj) [inline Source/DiscoveryV2/Models/QueryAggregation.swift]: Secret merge conflict!

File: Source/VisualRecognitionV4/Models/ImageDetails.swift [hunk 3]
```diff
@@ -24,17 +24,17 @@ public struct ImageDetails: Codable, Equatable {
     /**
      The identifier of the image.
      */
-    public var imageID: String
+    public var imageID: String?
```
Comment (mkistler) [inline Source/VisualRecognitionV4/Models/ImageDetails.swift]: This is also technically a breaking change, along with the properties below have been changed to optionals.  Are we okay with this?

File: Tests/AssistantV1Tests/AssistantTests.swift [hunk 4]
```diff
@@ -166,66 +160,9 @@ class AssistantTests: XCTestCase {
...
-
-            // verify output
-            XCTAssertTrue(message.output.logMessages.isEmpty)
-            XCTAssertEqual(message.output.text, result2)
-            XCTAssertNotNil(message.output.nodesVisited)
-            XCTAssertEqual(message.output.nodesVisited!.count, 3)
-
-            expectation2.fulfill()
-        }
-        waitForExpectations()
```
Comment (mkistler) [inline Tests/AssistantV1Tests/AssistantTests.swift]: Why are we deleting tests?

File: Tests/DiscoveryV2Tests/DiscoveryV2Tests.swift [hunk 5]
```diff
@@ -0,0 +1,828 @@
...
+        guard let url = bundle.url(forResource: name, withExtension: ext) else { return nil }
+        #endif
+        let data = try? Data(contentsOf: url)
+        return data
+    }
+    
+    // MARK: - Test Definition for Linux
+    
+    static var allTests: [(String, (DiscoveryTests) -> () throws -> Void)] {
+        let tests: [(String, (DiscoveryTests) -> () throws -> Void)] = []
```
Comment (mkistler) [inline Tests/DiscoveryV2Tests/DiscoveryV2Tests.swift]: Shouldn't you have the tests listed here to run on Linux?

File: Source/VisualRecognitionV4/Models/ImageDetails.swift (hunk 3 above)
Comment (mediumTaj) [reply Source/VisualRecognitionV4/Models/ImageDetails.swift]: Looks like these were intentionally marked as optional in the API definition. 

```
https://github.ibm.com/Watson/developer-cloud--api-definitions/pull/1488
```

Depending on the operation these fields in the response may not be returned.

File: Tests/DiscoveryV2Tests/DiscoveryV2Tests.swift (hunk 5 above)
Comment (repjarms) [reply Tests/DiscoveryV2Tests/DiscoveryV2Tests.swift]: We don't have a CPD environment capable of running tests on Linux (due to the fact that we do not support self signed certs in the Swift SDK for CPD)

File: Tests/AssistantV1Tests/AssistantTests.swift (hunk 4 above)
Comment (repjarms) [reply Tests/AssistantV1Tests/AssistantTests.swift]: This deleted a portion of a test that was very tightly coupled with a specific node in a test workspace that has been deleted. This makes the test more generic without hampering what we are really trying to test here.

File: Source/DiscoveryV2/Models/QueryAggregation.swift (hunk 2 above)
Comment (repjarms) [reply Source/DiscoveryV2/Models/QueryAggregation.swift]: I will fix and update the PR.

Comment (repjarms) [reply Source/DiscoveryV2/Models/QueryAggregation.swift]: resolved.

Comment (repjarms) [reply Source/DiscoveryV2/Models/QueryAggregation.swift]: resolved

File: Tests/DiscoveryV2Tests/DiscoveryV2Tests.swift (hunk 5 above)
Comment (mkistler) [reply Tests/DiscoveryV2Tests/DiscoveryV2Tests.swift]: Is Discovery CPD only?  I thought it was going to be available as a public service as well.

**Pull request merged**